src/components/vm/overview/watchdog.jsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cockpit from 'cockpit';
+import {
+    Button,
+    Flex, FlexItem,
+    Form, FormGroup,
+    Modal,
+    Radio,
+} from '@patternfly/react-core';
+import { useDialogs } from 'dialogs.jsx';
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { domainRemoveWatchdog, domainSetWatchdog } from "../../../libvirtApi/domain.js";
+import { rephraseUI } from "../../../helpers.js";
+import { WarningInactiveAlert, WarningInactiveTooltip } from "../../common/warningInactive.jsx";
+
+const _ = cockpit.gettext;
+
+export const WatchdogModal = ({ vm, isWatchdogAttached, idPrefix }) => {
+    const [dialogError, setDialogError] = useState();
+    const [watchdogAction, setWatchdogAction] = useState(vm.watchdog.action || "no-device");
+
+    const Dialogs = useDialogs();
+
+    const save = () => domainSetWatchdog({
+        connectionName: vm.connectionName,
+        vmName: vm.name,
+        hotplug: vm.state === "running",
+        permanent: vm.persistent,
+        action: watchdogAction,
+        isWatchdogAttached,
+    })
+            .then(Dialogs.close, exc => setDialogError({ text: _("Failed to configure watchdog"), detail: exc.message }));
+
+    const detach = () => domainRemoveWatchdog({
+        connectionName: vm.connectionName,
+        vmName: vm.name,
+        hotplug: vm.state === "running",
+        permanent: vm.persistent,
+        model: vm.watchdog.model,
+    })
+            .then(Dialogs.close, exc => setDialogError({ text: _("Failed to detach watchdog"), detail: exc.message }));
+
+    const supportedActions = ["reset", "poweroff", "inject-nmi", "pause"];
+
+    const showWarning = () => {
+        if (vm.state === 'running' && isWatchdogAttached && vm.watchdog.action !== watchdogAction)
+            return <WarningInactiveAlert idPrefix={idPrefix} />;
+    };
+
+    return (
+        <Modal id={`${idPrefix}-watchdog-modal`}
+               position="top"
+               variant="small"
+               onClose={Dialogs.close}
+               title={_("Watchdog device type")}
```

Comment:
Let's configure the title accordingly. When adding watchdig maybe "Add a watchdog device"or when editing/removing "Edit watchdog device"